client/termsetting/handlers/numeric.toggle.ts

```diff
@@ -69,6 +69,17 @@ export async function getHandler(self) {
 	}
 
 	if (self.opts.numericEditMenuVersion!.includes('discrete')) {
+		if (!self.term.bins && self.term.type !== 'survival') {
```

Comment:
getDefaultBins() is already called in `fillTW()` for non-dict terms.
when can bins be missing?

Reply:
@xzhou82 When using `fillTW()` from a `geneExpression` term.type handler, the `getCategories()` that gets called in `postMain()` slows down the loading of the plot by a lot. The solution that I suggested was to not request the default bins data until it is needed, in this case only when opening the edit UI.

Reply:
when tw.q.mode == 'continuous' its not called in fillTW(). Only called here when users click the term and the term has this discrete tab

Reply:
ok i see makes sense. may simply delete `getDefaultBins()` call in fillTW()? since it will be filled as needed when discrete tab will be used

Reply:
We tried that fix initially to not call `getDefaultBins()` at all, but barchart and maybe other plots that render discrete values require the samplecounts-by-category to be present when loading a plot. So the solution was to check to skip only if the mode is continuous.

Reply:
so this change won't cover all basis? could you think about a different way to centralize call of getDefaultBins? nice to get rid of the code repetition

Reply:
"so this change won't cover all basis" - Yes, the change only affects plots that render data with `tw.q.mode == 'continuous'`. The code block with `getDefaultBins()` in both `fillTW()` and tabCallback can be centralized as `handler.maySetBins()` and called in both functions to avoid repetition. But maybe do this in another PR?

Reply:
fine to do in another PR. thanks